lib/primitives/elementary/line_segment.rb

```diff
@@ -5,13 +5,31 @@ class LineSegment
       attr_reader :ends_at
 
       def initialize(begins_at, ends_at)
-        # Your code goes here...
+        check_is_a_point(begins_at)
+        check_is_a_point(ends_at)
+        @begins_at = begins_at
+        @ends_at = ends_at
       end
 
+      # Count a length of a line segment based on two points
       def length
-        # Your code goes here...
-        # REQUIREMENT: use lazy evaluation
+        begins_at.distance ends_at
       end
+
+      # Return straight line object based points coordinates
+      def to_straight_line
+        a = ends_at.y - begins_at.y
+        b = begins_at.x - ends_at.x
+        c = -begins_at.x * ends_at.y + begins_at.y * ends_at.x
+        StraightLine.new(a, b, c)
+      end
+
+      def check_is_a_point(x)
```

Comment:
This utility method may be [private](https://www.rubydoc.info/stdlib/core/Module:private).